tests/FrontBaseClass.php

```diff
@@ -61,18 +63,27 @@ public function afterTestMethod($method)
 
     protected function logIn()
     {
-        $crawler = $this->http_client->request('GET', $this->base_uri . 'index.php');
+
+        $this->http_client->request('GET', $this->base_uri . 'index.php');
+        $crawler = $this->http_client->waitFor('#login_name');
+
         $login_name = $crawler->filter('#login_name')->attr('name');
         $pass_name = $crawler->filter('input[type=password]')->attr('name');
-        $form = $crawler->selectButton('submit')->form();
-        $form[$login_name] = TU_USER;
-        $form[$pass_name] = TU_PASS;
-        //proceed form submission
-        $crawler = $this->http_client->submit($form);
+
+        $crawler = $this->http_client->submitForm(
+            'Sign in',
+            [
+                $login_name => TU_USER,
+                $pass_name => TU_PASS,
+            ]
+        );
 
         //once logged in, we reach standard interface
-        $page_title = $crawler->filter('title')->text();
-        $this->string($page_title)->isIdenticalTo('Standard interface - GLPI');
+        //$this->http_client->takeScreenshot('logged.png'); // see if that works...
+        //FIXME: should work :'(
+        //according to doc, we should rely on html() rather than text() for non-displayed elements
+        /*$page_title = $crawler->filter('title')->html();
+        $this->string($page_title)->isIdenticalTo('Standard interface - GLPI');*/
```

Comment:
You should wait for the page to load before checking the title.
$this->getClient()->waitFor('div.page');
$this->string($this->getClient()->getTitle())->isIdenticalTo('Standard interface - GLPI');

Worked for me.

Reply:
In facts, this is not the wait that solve the issue, but using the `getTitle()` method, thanks :)

Reply:
Ah. I was overly cautious. By coincidence, I was looking at this same test framework last night for use with GLPI. I added a few helper methods though to try and make it less verbose, unify the different data between the client and crawler, clarify the timeout exceptions, and work around not being able to input data directly for non-form inputs.

In case you want to consider them (I was using PHPUnit so it would be a little different):
```php
/** @var Client */
    private $client;

    private const DEFAULT_TIMEOUT = 1;

    protected function getClient(): Client
    {
        if ($this->client === null) {
            $this->client = static::createPantherClient();
        }
        return $this->client;
    }

    protected function visit(string $url, array $options = [])
    {
        $method = $options['method'] ?? 'GET';
        $this->getClient()->request($method, $url);
    }

    protected function title(): string
    {
        return $this->getClient()->getTitle();
    }

    protected function url(): string
    {
        return $this->getClient()->getCurrentURL();
    }

    protected function get(string $selector, array $options = [])
    {
        $this->expectToExist($selector, $options);
        return $this->getClient()->getCrawler()->filter($selector);
    }

    protected function setValue(string $selector, mixed $value, array $options = [])
    {
        $this->expectToExist($selector, $options);
        $val = is_string($value) ? "'$value'" : $value;
        $this->client->executeScript(
            "$('$selector').val($val);"
        );
    }

    protected function click(string $selector, array $options = [])
    {
        $this->expectToExist($selector, $options);
        $this->getClient()->getCrawler()->filter($selector)->click();
    }

    protected function expectToExist(string $selector, array $options = [])
    {
        $timeout = $options['timeout'] ?? self::DEFAULT_TIMEOUT;
        try {
            $this->getClient()->waitFor($selector, $timeout);
        } catch (TimeoutException) {
            $this->fail("Expected to find '$selector' within $timeout seconds");
        }
    }

    protected function screenshot(?string $name = null)
    {
        $this->getClient()->takeScreenshot($name);
    }
```